### ISSUE TITLE
Search Functionality for the Select Widget

### ISSUE THREAD

**Issue by davemagro** [Textualize/textual#4167]
I think this would be a very useful feature. Kind of needed in my current project as well. 

Leaving an issue to hopefully get back to this in the future. Hope I could find the time to implement this myself.

**Comment by fer-hnndz** [comment]
I would also like to start working on this feature. First of all, I believe it should be decided how would the search future work in detail.
Will you be able to search for an item by completely typing it or it will just jump to the first item with the letter on the list?

**Comment by TomJGooding** [comment]
I agree some sort of "autocomplete" widget would be extremely useful and something I've seen often requested. For example my own issue here:

- https://github.com/Textualize/textual/issues/3362

If someone does want to take this on, I think the core functionality for such a widget has already been worked out in the command palette if that helps.

**Comment by fer-hnndz** [comment]
I'll look intro the command palette code. I just realized it had that autocomplete function.
Not sure if it should be added as a whole new widget or as a feature for select widgets.

**Comment by TomJGooding** [comment]
I think this should be a new widget. A simple dropdown list is very different from a text input enhanced by suggested options. Not to mention the code for the `Select` widget is already complicated enough as it is!

But if you're serious about working on this, I'd wait for some input from a Textual maintainer.

**Comment by taleog** [comment]
Hi! We’re a group of software engineering students and would like to take on this issue as part of one of our courses. Could it be assigned to @taleog so we can work on it and avoid duplicate work?

Thank you!
